base_layer/core/src/validation/aggregate_body/aggregate_body_internal_validator.rs

```diff
@@ -143,6 +145,58 @@ impl AggregateBodyInternalConsistencyValidator {
     }
 }
 
+fn check_confidential_output_utxo(output: &TransactionOutput) -> Result<(), ValidationError> {
+    if let Some(conf_output) = output.features.confidential_output_data() {
+        if conf_output.network.is_some() || conf_output.network_knowledge_proof.is_some() {
+            // If one of these is set, both must be set
+            if conf_output.network.is_none() || conf_output.network_knowledge_proof.is_none() {
+                return Err(ValidationError::ConfidentialOutputNetworkNotSet);
+            }
+            // If set, the signature must be valid
+            let sig_pub_key = conf_output.network.as_ref().unwrap();
+            if !conf_output
+                .network_knowledge_proof
+                .as_ref()
+                .unwrap()
+                .verify(sig_pub_key, conf_output.claim_public_key.to_vec())
+            {
+                return Err(ValidationError::TemplateInvalidNetworkKnowledgeProof);
+            }
+        }
```

Comment:
nit: think we should dry out the sidechain network proof code

```suggestion
        if conf_output.network.is_some() || conf_output.network_knowledge_proof.is_some() {
            // If one of these is set, both must be set
            if conf_output.network.is_none() || conf_output.network_knowledge_proof.is_none() {
                return Err(ValidationError::ConfidentialOutputNetworkNotSet);
            }
            
               check_sidechain_network_proof(
                 conf_output.network.as_ref().unwrap(),
                 conf_output.network_knowledge_proof.as_ref().unwrap()
               )?;   
       }
```

Reply:
They return specific errors, but I suppose you could return the same error for each of them

Reply:
ah yeah, not too fussed I see why you did that